x/staging/subspaces/keeper/msg_server.go

```diff
@@ -0,0 +1,182 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/staging/subspaces/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the stored MsgServer interface
+// for the provided keeper
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (k msgServer) CreateSubspace(goCtx context.Context, msg *types.MsgCreateSubspace) (*types.MsgCreateSubspaceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check the if the subspace already exists
+	if k.DoesSubspaceExist(ctx, msg.SubspaceID) {
+		return nil,
+			sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "the subspaces with id %s already exists", msg.SubspaceID)
+	}
+
+	// Create and store the new subspaces
+	subspace := types.NewSubspace(msg.SubspaceID, msg.Name, msg.Creator, msg.Creator, msg.SubspaceType, ctx.BlockTime())
+
+	// Validate the subspace
+	if err := subspace.Validate(); err != nil {
+		return nil, err
+	}
+
+	_ = k.SaveSubspace(ctx, subspace, subspace.Owner)
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
+		types.EventTypeCreateSubspace,
+		sdk.NewAttribute(types.AttributeKeySubspaceID, msg.SubspaceID),
+		sdk.NewAttribute(types.AttributeKeySubspaceName, msg.Name),
+		sdk.NewAttribute(types.AttributeKeySubspaceCreator, msg.Creator),
```

Comment:
We should also add the creation time here since it's only retrievable from the block itself otherwise (which might not be handy for some clients like BDJuno)